packages/esm-patient-orders-app/src/order-cancellation-form/cancel-order-form.component.tsx

```diff
@@ -0,0 +1,182 @@
+import React, { useCallback, useMemo, useState } from 'react';
+import { Controller, useForm } from 'react-hook-form';
+import { useTranslation } from 'react-i18next';
+import { z } from 'zod';
+import { zodResolver } from '@hookform/resolvers/zod';
+import {
+  Button,
+  TextArea,
+  ButtonSet,
+  Column,
+  Form,
+  InlineNotification,
+  Stack,
+  DatePicker,
+  DatePickerInput,
+} from '@carbon/react';
+import { showSnackbar, useLayoutType } from '@openmrs/esm-framework';
+import { usePatientOrders, type Order } from '@openmrs/esm-patient-common-lib';
+import styles from './cancel-order-form.scss';
+import { saveCancelOrderRequest } from './cancel-order.resource';
+import dayjs from 'dayjs';
+
+interface OrderCancellationFormProps {
+  order: Order;
+  patientUuid: string;
+  context?: string;
+  closeWorkspace: () => void;
+}
+
+const OrderCancellationForm: React.FC<OrderCancellationFormProps> = ({ order, patientUuid, closeWorkspace }) => {
+  const { t } = useTranslation();
+  const isTablet = useLayoutType() === 'tablet';
+  const [isSubmitting, setIsSubmitting] = useState(false);
+  const [showErrorNotification, setShowErrorNotification] = useState(false);
+
+  const { mutate } = usePatientOrders(patientUuid, 'ACTIVE');
+
+  const cancelOrderSchema = useMemo(() => {
+    return z.object({
+      cancellationDate: z.date(),
+      reasonForCancellation: z.string(),
+    });
+  }, []);
+
+  type CancelOrderFormData = z.infer<typeof cancelOrderSchema>;
+
+  const { control, handleSubmit, watch, setValue } = useForm<CancelOrderFormData>({
```

Comment:
Thank you for using the React hook form.

I am currently working on #1613, which uses the `promptBeforeClosing` in the workspaces.

Please add the following snippet in the code (and please take care of the same in future implementations)

```
const OrderCancellationForm: React.FC<OrderCancellationFormProps> = ({ 
	order, 
	patientUuid, 
	closeWorkspace,
	promptBeforeClosing, 
}) => {
....
const { control, handleSubmit, watch, setValue, formState: { isDirty } } = useForm<CancelOrderFormData>({
...

useEffect(() => {
	promptBeforeClosing(() => isDirty);
}. [isDirty])
```

Feel free to refer the PR attached.
Thanks!